src/Controller/AppController.php

```diff
@@ -98,6 +99,21 @@ public function initialize()
         $this->Crud->addListener('relatedModels', 'Crud.RelatedModels');
     }
 
+    public function customLog($description) {
+        // log this
+        $this->logs = TableRegistry::getTableLocator()->get('Logs');
+        $log = $this->logs->newEntity();
+        $log->description   = $description;
+        $log->user          = $this->Auth->user()['samaccountname'];
+        $log->date_time     = date('Y-m-d H:i:s');
+        $log->ip_address    = $_SERVER['REMOTE_ADDR'];
```

Comment:
This can potentially be wrong (show the IP of the load balancer or proxy), it should probably be improved to use HTTP_X_FORWARDED_FOR if available.